src/job_runner.cc

```diff
@@ -54,6 +58,68 @@ class JobRunnerWorker {
     static constexpr std::size_t kInitialQueueCapacity = 8192;
 };
 
+class JobRunnerStrand : public std::enable_shared_from_this<JobRunnerStrand> {
+   public:
+    using job_t       = JobRunner::job_t;
+    using job_queue_t = boost::lockfree::queue<job_t*>;
+
+    JobRunnerStrand(std::weak_ptr<JobRunner> runner) : runner_weak_(runner) {}
```

Comment:
1. explicit?
2. Put it in cpp, header change is expensive w.r.t. compile time.

Reply:
> Put it in cpp, header change is expensive w.r.t. compile time.

This is not a header :)

Reply:
Because there are no need to know this class externally (other than the declaration). If we unit-test, we test the runner and see if the strand gives us serialization and that is it.